### ISSUE TITLE
Using Email Channel SPAM replies

### ISSUE THREAD

**Issue by DanteNahuel** [Microsoft/botbuilder-dotnet#1331]
Hi
When replying to an email received through the email channel, the bot sends over 400 replies, spamming the inbox. I'm not sure why this happens although it seems like once one email is sent, the code enters a loop of replying and receiving emails and never ends.
I'm using SDK 4 .Net Core, running the code over Visual Studio Locally, using ngrok to debug locally while publishing to Azure Bot Channel portal.
Here's the code:

```

private static async Task<DialogTurnResult> CheckQuestionAsync(
        WaterfallStepContext stepContext,
        CancellationToken cancellationToken = default(CancellationToken))
        {
            
             if (stepContext.Context.Activity.ChannelId == "email")
            {
            await stepContext.Context.SendActivityAsync("test");
                               
            return await stepContext.EndDialogAsync();
        }
        }




public async Task OnTurnAsync(ITurnContext turnContext, CancellationToken cancellationToken = default(CancellationToken))
        {
            if (turnContext.Activity.Type == ActivityTypes.Message )
            {

                
                // Establish dialog state from the conversation state.
                DialogContext dc = await _dialogs.CreateContextAsync(turnContext, cancellationToken);

                // Get the user's info.
                UserInfo userInfo = await _accessors.UserInfoAccessor.GetAsync(turnContext, () => new UserInfo(), cancellationToken);

                await _accessors.UserInfoAccessor.SetAsync(turnContext, userInfo, cancellationToken);

                // Continue any current dialog.
                DialogTurnResult dialogTurnResult = await dc.ContinueDialogAsync();
                
                // Process the result of any complete dialog.
                if (dialogTurnResult.Status is DialogTurnStatus.Complete)
                {
                    var i = 0;
                   
                    }
                }

                // Every dialog step sends a response, so if no response was sent,
                // then no dialog is currently active.
                
                else if (!turnContext.Responded)
                {
                    var h = 0;
                    
                        //We need to think how to handle too many attemps.
                        await dc.BeginDialogAsync(MainDialogId, null, cancellationToken);
                 
                }

                // Save the new turn count into the conversation state.
                await _accessors.ConversationState.SaveChangesAsync(turnContext, false, cancellationToken);
                await _accessors.UserState.SaveChangesAsync(turnContext, false, cancellationToken);
            }
           
         
            else
            {
                // Commenting this to avoid "event detected" message over the chat.
               // await turnContext.SendActivityAsync($"{turnContext.Activity.Type} event detected");
            }
        }



```

**Comment by DanteNahuel** [comment]
I keep receiving the same reply over and over again. It's been half an hour since I disabled the email channel in Azure and stopped Visual Studio + ngrok.
I don't know what makes this loop but i'm not able to stop it and it's probably going to get the outlook email account blocked.

**Comment by DanteNahuel** [comment]
Over 1 hour later and I kept receiving emails. I checked the email account and it was actively sending the reply over and over again, even after deleting the Email channel from the Bot Channel Azure configuration panel.
I had to delete the Bot Channel Registration resource in order to make it stop.
I believe this a bug and a very important once since it can make your bot spam emails like Mickey mouse  can spam brooms, although I'm not sure what in my code causes it.
